coderd/autobuild/lifecycle_executor.go

```diff
@@ -266,6 +334,26 @@ func isEligibleForAutostop(build database.WorkspaceBuild, job database.Provision
 		!currentTick.Before(build.Deadline)
 }
 
+// isEligibleForLockedStop returns true if the workspace should be locked
+// for breaching the inactivity threshold of the template.
+func isEligibleForLockedStop(ws database.Workspace, templateSchedule schedule.TemplateScheduleOptions) bool {
+	// Only attempt to lock workspaces not already locked.
+	return !ws.LockedAt.Valid &&
+		// The template must specify an inactivity TTL.
+		templateSchedule.InactivityTTL > 0 &&
+		// The workspace must breach the inactivity TTL.
+		database.Now().Sub(ws.LastUsedAt) > templateSchedule.InactivityTTL
```

Comment:
It's likely preferable to use `currentTick` instead here. Same for `isEligibleForDelete`.